### PR TITLE
Timeline duration support. Updated most of abilities, but not sure ab…

### PR DESCRIPTION
**Pull request by Airex**
…out some, as do not play those classes fulltime.

### CODE REVIEW

**Comment by ayyaruq** [comment]
For the DoTs, if they're on GCD that might get a bit weird. Is it possible to use the job colours, or a differentiator for damage vs heals/shields?

**Comment by Airex** [comment]
> For the DoTs, if they're on GCD that might get a bit weird. Is it possible to use the job colours, or a differentiator for damage vs heals/shields?

This implementation for ogcd abilities and effective when duration <= cooldown to show actual time cooldown is working

For gcd ones it is better to expand table with additional status row to show actual duration of applied effect

**Comment by ayyaruq** [comment]
I thought Windbite on BRD was a GCD 🤔 

Still wondering about colours for the fills tho, going by the code and your screenshots in discord.

**Comment by Airex** [comment]
> I thought Windbite on BRD was a GCD 🤔
> 
> Still wondering about colours for the fills tho, going by the code and your screenshots in discord.

Any proposals for colors? And in this case we need some way to identify what color to use, should it be property on ability?

**Comment by ayyaruq** [comment]
Job colours I guess, they're in `data/JOBS`. Could maybe use a fill for regular actions, and either light blue/yellow for heals/shields, or just lower transparency or something.

**Comment by Airex** [comment]
From my experience abilities with active duration could be splitted into types as:
Self defense - light blue
Party Defense- darker blue
Self damage buff - light red
Party damage buff - darker red
Healing buff - light green

Does this make sense?

**Comment by Airex** [comment]
Anything from my side should be done here?

**Comment by AzariahAeroborn** [comment]
Sorry for the delay in getting back to you.

Based on the conversation Ackwell and I had on Discord last Wednesday, I'm going to update #594 to switch the order of the linking between Actions and Statuses (so that Actions have a property for which Statuses they generate).  Once that's in place, you should be able to use that value to look up the status and get the duration from the status data.  

I should have the PR updated later today, can't guarantee when it's going to be merged in and available.  If you have any thoughts on how you want those linkages to map up, let me know.

**Comment by AzariahAeroborn** [comment]
The links from ACTIONS to the STATUSES they grant should be in place. If you notice any that are missing or incorrect, let me know or feel free to correct them. I’m happy to talk on discord if you have any questions on what I did.

**Comment by Airex** [comment]
I have reviewed approach to show statuses in timeline. This approach depends on statuses attached to actions and uses fflogs event to populate statuses with exact values of durations based on these events.

File: src/data/ACTIONS/MCH.js [hunk 1]
```diff
@@ -230,6 +230,7 @@ export default {
 		icon: 'https://xivapi.com/i/003000/003501.png',
 		onGcd: false,
 		cooldown: 6,
+		duration: 20,
```
Comment (AzariahAeroborn) [inline src/data/ACTIONS/MCH.js]: My understanding is that AQ is more in line with DRK's pet, where it just does its own thing independent of the machinist, and is not a damage amp to the caster's GCDs like NIN's Bunshin does, and AQ does not give a buff.  Does it make sense to show it as a buff on the timeline?  If we're going to do that, should we do the same thing with DRK's Living Shadow as well?

File: src/data/ACTIONS/ROG.js [hunk 2]
```diff
@@ -98,6 +98,7 @@ export default {
 		icon: 'https://xivapi.com/i/000000/000618.png',
 		onGcd: false,
 		cooldown: 60,
+		duration: 10,
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/ROG.js]: done

File: src/data/ACTIONS/ROG.js [hunk 3]
```diff
@@ -106,5 +107,6 @@ export default {
 		icon: 'https://xivapi.com/i/000000/000607.png',
 		onGcd: false,
 		cooldown: 120,
+		duration: 20,
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/ROG.js]: done

File: src/data/ACTIONS/PLD.js [hunk 4]
```diff
@@ -164,13 +165,15 @@ export default {
...
 		cooldown: 180,
+		statusesApplied: [STATUSES.SENTINEL],
 	},
 	CIRCLE_OF_SCORN: {
 		id: 23,
 		name: 'Circle Of Scorn',
 		icon: 'https://xivapi.com/i/000000/000161.png',
 		onGcd: false,
 		cooldown: 25,
+		statusesApplied: [STATUSES.CIRLCE_OF_SCORN],
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/PLD.js]: done

File: src/data/ACTIONS/NIN.js [hunk 5]
```diff
@@ -211,5 +212,6 @@ export default {
 		icon: 'https://xivapi.com/i/002000/002927.png',
 		onGcd: false,
 		cooldown: 110,
+		duration: 15,
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/NIN.js]: done

File: src/data/ACTIONS/NIN.js [hunk 6]
```diff
@@ -162,6 +162,7 @@ export default {
 		icon: 'https://xivapi.com/i/002000/002918.png',
 		onGcd: false,
 		cooldown: 60,
+		duration: 10,
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/NIN.js]: done

File: src/data/ACTIONS/BLM.js [hunk 7]
```diff
@@ -155,6 +155,7 @@ export default {
 		name: 'Manaward',
 		icon: 'https://xivapi.com/i/000000/000463.png',
 		cooldown: 120,
+		duration: 20,
```
[the remark this answers is not in the document]
Comment (Airex) [reply src/data/ACTIONS/BLM.js]: done

File: src/data/ACTIONS/MCH.js (hunk 1 above)
Comment (Airex) [reply src/data/ACTIONS/MCH.js]: this question is tricky, I can add functionality to generate such statuses, just for information purposes

**Comment by Airex** [comment]
I am going to add more statuses and link them to actions with next PRs. Or even other contributors can help me with this.

**Pull request closed without merging**